File: src/main/java/io/github/jho951/assignment/config/WorkerProperties.java
package io.github.jho951.assignment.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "worker")
public record WorkerProperties(
        String baseUrl,
        String issueKeyPath,
        String processPath,
        long timeoutMs,
        String candidateName,
        String candidateEmail
) {}

File: src/main/kotlin/io/github/jho951/assignment/config/WorkerProperties.kt
package io.github.jho951.assignment.config

import kotlin.jvm.JvmRecord
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "worker")
@JvmRecord
data class WorkerProperties(
    val baseUrl: String,
    val issueKeyPath: String,
    val processPath: String,
    val timeoutMs: Long,
    val candidateName: String,
    val candidateEmail: String
)
